dispenser/base/ModBusDispenser.ts

```diff
@@ -178,33 +232,72 @@ export class ModBusDispenser implements IDispenser {
         return parseInt(hexPair, 16); // Use parseInt for hex conversion
     }
 
-    async writeTotalizerToFile (datObj: TotalizerResponse): Promise<void> {
-        const filename = (await this.connection).totalizerFile;
-        if (!filename) {
-            throw new Error('Filename is undefined or empty');
-        }
-        const data = JSON.stringify(datObj, function (key, value) {
-            if (typeof value === 'bigint') {
-              return value.toString();
-            } else {
-              return value;
-            }
+    async writeTotalizerToFile(datObj: TotalizerResponse, orderCode: number, customerAssetId: string, sessionId: string): Promise<void> {
+
+        const db = await open({
+            filename: path.join(__dirname, 'dispenser.db'),
+            driver: sqlite.Database
         });
-        await fs.writeFile(filename, data);
-        debugLog('Successfully wrote object to file: %o', data);
+    
+        try {   
+            await db.run(
+                `INSERT INTO totalizer (order_code, customer_asset_id, session_id, totalizer, batchNumber, timestamp) 
+                 VALUES (?, ?, ?, ?, ?, ?)`,
+                orderCode, 
+                customerAssetId, 
+                sessionId,
+                datObj.totalizer, 
+                datObj.batchNumber?.toString(), 
+                datObj.timestamp 
+            );
+            
+            debugLog('Successfully wrote object to database: %o', datObj);
+            await this.clearOldTotalizerRecords();
+        } catch (error) {
+            console.error('Error writing totalizer to database:', error);
```

Comment:
Why not throwing any error to inform the application calling this method that an error has occurred?